### DISCUSSION TITLE
Support multiple spec files / selector to navigate multiple files

### DISCUSSION THREAD

**Discussion by geekbleek** [scalar/scalar#1019] [Ideas]
Love the project - it is looking really good!

It would be great to have a selector / be able to load multiple spec files into the UI.  I only tried the CDN method, but the first one loaded won out.  Bonus points if you could load the folder names each spec file is in for navigation!

Alternatively, could you combine multiple spec files into one view?  We have many different services generating their own specs and would like to consolidate them into one simple view.

Thanks for the great solution can't wait to see how this develops!

**Comment by hanspagel** [comment]
Ha! We’re working on exactly this right now. It’s a bigger effort though, because we’re building a new foundation for our packages: https://github.com/scalar/openapi-parser/tree/chore/new-package-api

As far as I know, this is going to be the first OpenAPI parser, that works with references to other files *in the browser*. 👀

**Comment by immanuelfodor** [comment]
I see this as a great way to add API versioning capabilities. One API, multiple API versions with their related OAS. Bonus point if it can be selected from a dropdown.

**Comment by hanspagel** [comment]
works for more than a year already :)
